library/Freckle/App/Kafka.hs

```diff
@@ -40,37 +43,58 @@ readKafkaBrokerAddresses t = case NE.nonEmpty $ T.splitOn "," $ T.pack t of
     | x /= "" -> Right $ BrokerAddress <$> xs
   _ -> Left "Broker Address cannot be empty"
 
-data KafkaProducerPool
-  = NullKafkaProducerPool
-  | KafkaProducerPool (Pool KafkaProducer)
-
-class HasKafkaProducerPool env where
-  kafkaProducerPoolL :: Lens' env KafkaProducerPool
-
-instance HasKafkaProducerPool site => HasKafkaProducerPool (HandlerData child site) where
-  kafkaProducerPoolL = envL . siteL . kafkaProducerPoolL
-
-createKafkaProducerPool
-  :: NonEmpty BrokerAddress
-  -> Int
+data KafkaProducerPoolConfig = KafkaProducerPoolConfig
+  { kafkaProducerPoolConfigStripes :: Int
   -- ^ The number of stripes (distinct sub-pools) to maintain.
   -- The smallest acceptable value is 1.
-  -> NominalDiffTime
+  , kafkaProducerPoolConfigIdleTimeout :: NominalDiffTime
   -- ^ Amount of time for which an unused resource is kept open.
   -- The smallest acceptable value is 0.5 seconds.
   --
   -- The elapsed time before destroying a resource may be a little
   -- longer than requested, as the reaper thread wakes at 1-second
   -- intervals.
-  -> Int
+  , kafkaProducerPoolConfigSize :: Int
   -- ^ Maximum number of resources to keep open per stripe.  The
   -- smallest acceptable value is 1.
   --
   -- Requests for resources will block if this limit is reached on a
   -- single stripe, even if other stripes have idle resources
   -- available.
+  }
+  deriving stock (Show)
+
+-- Same defaults as `Database.Persist.Sql.ConnectionPoolConfig`
+defaultKafkaProducerPoolConfig :: KafkaProducerPoolConfig
+defaultKafkaProducerPoolConfig = KafkaProducerPoolConfig 1 600 10
+
+envKafkaProducerPoolConfig
+  :: Env.Parser Env.Error KafkaProducerPoolConfig
+envKafkaProducerPoolConfig = do
+  poolSize <- Env.var Env.auto "KAFKA_PRODUCER_POOL_SIZE" $ Env.def 10
+  pure $ defaultKafkaProducerPoolConfig {kafkaProducerPoolConfigSize = poolSize}
```

Comment:
Whoa, we got some `ApplicativeDo` 'round here?

Reply:
`Parser` is only `Applicative`, so in order to write monadic looking parsers you have to enable it. I think we have it in some other `Parser` in `core` too.